tests/Microsoft.Identity.Test.Unit.net45/CacheTests/CacheSerializationTests.cs

```diff
@@ -0,0 +1,608 @@
+        public void TestSerializeMsalAccountCacheItem()
+        {
+            var item = CreateAccountItem();
+            string asJson = item.ToJsonString();
+            var item2 = MsalAccountCacheItem.FromJsonString(asJson);
+
+            AssertAccountCacheItemsAreEqual(item, item2);
+        }
+
+        [TestMethod]
+        public void TestSerializeMsalAccountCacheItemWithAdditionalFields()
+        {
+            var item = CreateAccountItem();
+
+            // Add an unknown field into the json
+            var asJObject = item.ToJObject();
+            asJObject["unsupported_field_name"] = "this is a value";
+
+            // Ensure unknown field remains in the AdditionalFieldsJson block
+            var item2 = MsalAccountCacheItem.FromJObject(asJObject);
+            Assert.AreEqual("{\r\n  \"unsupported_field_name\": \"this is a value\"\r\n}", item2.AdditionalFieldsJson);
+
+            // Ensure additional fields make the round trip into json
+            asJObject = item2.ToJObject();
+            AssertAccountHasJObjectFields(
+                asJObject,
+                new List<string>
+                {
+                    "unsupported_field_name"
+                });
+        }
+
+        [TestMethod]
+        public void TestMsalAccountCacheItem_HasProperJObjectFields()
+        {
+            var item = CreateAccountItem();
+            var asJObject = item.ToJObject();
+
+            AssertAccountHasJObjectFields(asJObject);
+        }
+
+        #endregion // ACCOUNT TESTS
+
+        #region DICTIONARY SERIALIZATION TESTS
+
+        [TestMethod]
+        public void TestDictionarySerialization()
+        {
+            var accessor = CreateTokenCacheAccessor();
+
+            var s1 = new TokenCacheDictionarySerializer(accessor);
+            byte[] bytes = s1.Serialize();
+            string json = new UTF8Encoding().GetString(bytes);
+
+            // TODO(cache): assert json value?  or look at JObject?
+
+            var otherAccessor = new TokenCacheAccessor();
+            var s2 = new TokenCacheDictionarySerializer(otherAccessor);
+            s2.Deserialize(bytes);
+
+            AssertAccessorsAreEqual(accessor, otherAccessor);
+        }
+
+        #endregion // DICTIONARY SERIALIZTION TESTS
+
+        #region JSON SERIALIZATION TESTS
+
+        [TestMethod]
+        public void TestJsonSerialization()
+        {
+            var accessor = CreateTokenCacheAccessor();
+
+            var s1 = new TokenCacheJsonSerializer(accessor);
+            byte[] bytes = s1.Serialize();
+            string json = new UTF8Encoding().GetString(bytes);
+
+            var otherAccessor = new TokenCacheAccessor();
+            var s2 = new TokenCacheJsonSerializer(otherAccessor);
+            s2.Deserialize(bytes);
+
+            AssertAccessorsAreEqual(accessor, otherAccessor);
+        }
+
+        #endregion // JSON SERIALIZATION TESTS
+
+        [TestMethod]
+        [DeploymentItem(@"Resources\cachecompat_python.bin")]
```

Comment:
>cachecompat_python.bin [](start = 36, length = 22)

seems like we should have a file from python both with items in it and one with-out items in it?

Reply:
it's only this way for now because I can't parse their file due to the array of scopes (versus a string of scopes).  i'll fix up and finalize this test once python gets updated.